pyomo/contrib/parmest/parmest.py

```diff
@@ -687,43 +667,41 @@ def _Q_at_theta(self, thetavals, initialize_parmest_model=False):
             # create dictionary to store pyomo model instances (scenarios)
             scen_dict = dict()
 
+        # keep track of parameter names as we enumerate models
+        param_names = []
         for snum in scenario_numbers:
             sname = "scenario_NODE" + str(snum)
             instance = _experiment_instance_creation_callback(sname, None, dummy_cb)
-            model_theta_names = self._expand_indexed_unknowns(instance)
+            model_param_list = _expand_indexed_unknowns(instance)
+            param_names.extend([c.name for c in model_param_list])
 
             if initialize_parmest_model:
                 # list to store fitted parameter names that will be unfixed
                 # after initialization
-                theta_init_vals = []
-                # use appropriate theta_names member
-                theta_ref = model_theta_names
+                param_init_vals = []
 
-                for i, theta in enumerate(theta_ref):
-                    # Use parser in ComponentUID to locate the component
-                    var_cuid = ComponentUID(theta)
-                    var_validate = var_cuid.find_component_on(instance)
+                for var_validate in model_param_list:
                     if var_validate is None:
                         logger.warning(
-                            "theta_name %s was not found on the model", (theta)
+                            "param %s was not found on the model", (var_validate.name)
```

Comment:
May be worth turning this into an `f`-string.
```suggestion
                            f"param {var_validate.name} was not found on the model"
```